File: eAnimalShelter/eAnimalShelter.Services/Database/AnimalBreed.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eAnimalShelter.Services.Database
{
    public class AnimalBreed
    {
        [Key]
        public int BreedId { get; set; }
        public string BreedName { get; set; } = string.Empty;
        public int SpeciesId { get; set; }

        [ForeignKey(nameof(SpeciesId))]
        public AnimalSpecies Species { get; set; } = null!;

    
        
    }
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/animal_breed.dart
class AnimalBreed {
  int breedId;
  String breedName;

  int speciesId;

  AnimalBreed({
    required this.breedId,
    required this.breedName,
    required this.speciesId,
  });

  factory AnimalBreed.fromJson(
    Map<String, dynamic> json,
  ) {
    return AnimalBreed(
      breedId: json["breedId"],
      breedName:
          json["breedName"] ?? "",
      speciesId:
          json["speciesId"] ?? 0,
    );
  }
}